pkg/trie/triedb/triedb.go

```diff
@@ -29,14 +29,33 @@ type TrieDB struct {
 	rootHash common.Hash
 	db       db.DBGetter
 	cache    cache.TrieCache
+	layout   trie.TrieLayout
+	// rootHandle is an in-memory-trie-like representation of the node
+	// references and new inserted nodes in the trie
+	rootHandle NodeHandle
+	// Storage is an in memory storage for nodes that we need to use during this
+	// trieDB session (before nodes are committed to db)
+	storage NodeStorage
+	// deathRow is a set of nodes that we want to delete from db
+	deathRow map[common.Hash]interface{}
```

Comment:
love this one!

Reply:
From looking at the references to this attribute, I only see writes to `deathRow` but no reads.  Is this required for something else?

Reply:
It will be use during [commit phase](https://github.com/paritytech/trie/blob/c3348e4b335443c17586a8b7491ba893cfde76a6/trie-db/src/triedbmut.rs#L1809-L1816)
If you want I can remove it from this PR and add it later in the PR implementing db commits

Reply:
I'm ok either way.